addons/website/static/src/js/widgets/theme.js

```diff
@@ -705,17 +675,36 @@ var ThemeCustomizeDialog = Dialog.extend({
     _updateStyle: function (enable, disable, reload) {
         var self = this;
 
-        var $loading = $('<i/>', {class: 'fa fa-refresh fa-spin'});
-        this.$modal.find('.modal-title').append($loading);
-
         if (reload || config.isDebug('assets')) {
-            window.location.href = $.param.querystring('/website/theme_customize_reload', {
-                href: window.location.href,
-                enable: (enable || []).join(','),
-                disable: (disable || []).join(','),
-                tab: this.$('.nav-link.active').parent().index(),
+            return new Promise((resolve, reject) => {
+                const confirm = Dialog.confirm(self, _t("This change needs to reload the page, this will save all your changes and reload the page, are you sure you want to proceed?") +
+                    (config.isDebug('assets') ? _t(" It appears you are in debug=assets mode, all theme customization options require a page reload in this mode.") : ''), {
+                    confirm_callback: () => resolve(true),
+                });
+                confirm.on('closed', self, () => resolve(false));
+            }).then((save) => {
+                if (save) {
+                    this._rpc({
+                        route: '/website/theme_customize',
+                        params: {
+                            'enable': enable,
+                            'disable': disable,
+                            'get_bundle': true,
+                        },
+                    }).then(() => {
+                        self.trigger_up('request_save', {
+                            reload: false,
+                            onSuccess: function () {
+                                window.location.href = window.location.origin + window.location.pathname + '?enable_editor=1';
+                            },
+                        });
+                    });
```

Comment:
Maybe it would be safer to save then to call theme_customize as some customize option may toggle views which would xpath / remove editable areas being edited.